spec/class-model.spec.ts

```diff
@@ -515,6 +519,27 @@ describe("class models", () => {
     });
   });
 
+  describe("with optional date properties", () => {
+    test("should be able to create read-only instances with numbers", () => {
+      const createdDate = new Date();
+      const v = NestedComplex.createReadOnly({ stringArray: [], numberMap: {}, createdDate: createdDate.getTime() });
+      expect(v.createdDate).toBeInstanceOf(Date);
+      expect(v.createdDate).toEqual(createdDate);
+    });
+
+    test("should be able to create read-only instances with date instances", () => {
+      const createdDate = new Date();
+      const v = NestedComplex.createReadOnly({ stringArray: [], numberMap: {}, createdDate });
+      expect(v.createdDate).toBeInstanceOf(Date);
+      expect(v.createdDate).toEqual(createdDate);
+    });
+
+    test("should be able to create read-only instances with undefined value", () => {
+      const v = NestedComplex.createReadOnly({ stringArray: [], numberMap: {} });
+      expect(v.createdDate).toBeInstanceOf(Date);
+    });
+  });
```

Comment:
Do these test cases pass? I think it flows through a slightly different path but while we're in here we should make sure it directly assigns null ok too

```typescript
    @register
    class NullyDate extends ClassModel({
      createdDate: types.maybeNull(types.Date),
    }) {}

    test("should be able to create read-only instances with nullable dates set to null", () => {
      const v = NullyDate.createReadOnly({ createdDate: null });
      expect(v.createdDate).toBeNull();
    });

   test("should be able to create read-only instances with nullable dates set to null", () => {
      const createdDate = new Date();
      const v = NullyDate.createReadOnly({ createdDate });
      expect(v.createdDate).toEqual(createdDate);
    });
```